### PR TITLE
bpfman: Add support for fentry/fexit program types

### PR DESCRIPTION
**Pull request by Billy99**
Add support for the fentry and fexit program types.

Resolves: #1012

### CODE REVIEW

File: bpfman/src/bpf.rs [hunk 1]
```diff
@@ -658,6 +664,52 @@ impl BpfManager {
...
+                    .load(&fn_name, &btf)
+                    .map_err(BpfmanError::BpfProgramError)?;
+                program.get_data_mut().set_kernel_info(&fentry.info()?)?;
+
+                let id = program.data.get_id()?;
+                let link_id = fentry.attach()?;
+                let owned_link: FEntryLink = fentry.take_link(link_id)?;
+                let fd_link: FdLink = owned_link.into();
+                fd_link
+                    .pin(format!("{RTDIR_FS}/prog_{}_link", id))
```
Comment (astoycos) [inline bpfman/src/bpf.rs:681]:
```suggestion
                    .pin(format!("{RTDIR_FS}/prog_{id}_link"))
```

Comment (astoycos) [reply bpfman/src/bpf.rs:681]: same below

File: tests/integration-test/src/tests/utils.rs [hunk 2]
```diff
@@ -460,6 +464,66 @@ pub fn add_kretprobe(
     Ok(prog_id)
 }
 
+/// Install a fentry program with bpfman
+pub fn add_fentry(load_type: &LoadType, image_url: &str, file_path: &str) -> Result<String> {
```
Comment (astoycos) [inline tests/integration-test/src/tests/utils.rs:468]: Nit but can you just collapse these into a single function with a boolean toggle or something? We already have so much duplicated code here. :/

**Review by astoycos — APPROVED** [review]
Just some small nits, we'll make the integration test program a bit better in a follow up, thanks @Billy99!!!

**Comment by astoycos** [comment]
(just remove hold when ready)

File: bpfman/src/bpf.rs [hunk 3]
```diff
@@ -337,6 +337,10 @@ impl BpfManager {
                 if let Some(pin_path) = program.get_data().get_map_pin_path()? {
                     let _ = self.cleanup_map_pin_path(&pin_path, map_owner_id);
                 }
+
+                // Cleanup and program that failed to create. Ignore any delete errors.
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfman/src/bpf.rs]: Done

File: bpfman/src/bpf.rs (hunk 1 above)
Comment (Billy99) [reply bpfman/src/bpf.rs:681]: Done.
There are multiple places that the variable is in the string, why is this method preferred?

File: tests/integration-test/src/tests/utils.rs (hunk 2 above)
Comment (Billy99) [reply tests/integration-test/src/tests/utils.rs]: Done

**Comment by Billy99** [comment]
I squashed the integration tests in with the main fentry/fexit code (because I addressed one of the comments in the wrong commit and just made it easier to combine them than to try to unwind the update).

File: bpfman/src/bpf.rs (hunk 1 above)
Comment (astoycos) [reply bpfman/src/bpf.rs:681]: If it's already in the right format it's just a bit less code :) but really it's just a style thing

**Pull request merged**